auth 4.0.4 libatomic configure error on ppc64

Full buildlog:
https://kojipkgs.fedoraproject.org//work/tasks/5577/20115577/build.log

checking whether the linker accepts -latomic... no
configure: error: Unable to link against libatomic, cannot continue

Any ideas?

Note: This issue affects only ppc64

@pieterlexis 

I have reverted commit 03571f7ac3d5bebb4879849b094e2e03f019cd10 but now it fails to build because of configure: error: ldap library (libldap) not found...

--- a/m4/pdns_check_os.m4
+++ b/m4/pdns_check_os.m4
@@ -36,7 +36,7 @@ AC_DEFUN([PDNS_CHECK_OS],[
   AM_CONDITIONAL([HAVE_SOLARIS], [test "x$have_solaris" = "xyes"])
 
   case "$host" in
-  mips* | powerpc* )
+  mips* | powerpc-* )
     AC_MSG_CHECKING([whether the linker accepts -latomic])
     LDFLAGS="-latomic $LDFLAGS"
     AC_LINK_IFELSE([m4_default([],[AC_LANG_PROGRAM()])],

